cylc/flow/scripts/lint.py

```diff
@@ -0,0 +1,502 @@
+    """Take a ruleset name and return a ruleset code
+
+    Examples:
+        >>> get_checkset_from_name('728')
+        ['U']
+        >>> get_checkset_from_name('style')
+        ['S']
+        >>> get_checkset_from_name('all')
+        ['U', 'S']
+    """
+    if name == '728':
+        purpose_filters = ['U']
+    elif name == 'style':
+        purpose_filters = ['S']
+    else:
+        purpose_filters = ['U', 'S']
+    return purpose_filters
+
+
+def parse_checks(check_arg):
+    """Collapse metadata in checks dicts.
+    """
+    parsedchecks = {}
+    purpose_filters = get_checkset_from_name(check_arg)
+
+    checks = {'U': get_upgrader_info(), 'S': STYLE_CHECKS}
+
+    for purpose, ruleset in checks.items():
+        if purpose in purpose_filters:
+            for index, (pattern, meta) in enumerate(ruleset.items(), start=1):
+                meta.update({'purpose': purpose})
+                meta.update({'index': index})
+                parsedchecks.update({pattern: meta})
+    return parsedchecks
+
+
+def check_cylc_file(file_, checks, modify=False):
+    """Check A Cylc File for Cylc 7 Config"""
+    # Set mode as read-write or read only.
+    outlines = []
+
+    # Open file, and read it's line to memory.
+    lines = file_.read_text().split('\n')
+    jinja_shebang = lines[0].strip().lower() == JINJA2_SHEBANG
+    count = 0
+    for line_no, line in enumerate(lines, start=1):
+        for check, message in checks.items():
+            # Tests with for presence of Jinja2 if no shebang line is
+            # present.
+            if (
+                jinja_shebang
+                and message['short'].startswith(
+                    JINJA2_FOUND_WITHOUT_SHEBANG)
+            ):
+                continue
+
+            if check.findall(line) and not line.strip().startswith('#'):
+                count += 1
+                if modify:
+                    if message['url'].startswith('http'):
+                        url = message['url']
+                    else:
+                        url = URL_STUB + message['url']
+                    outlines.append(
+                        f'# [{message["purpose"]}{message["index"]:03d}]: '
+                        f'{message["short"]}\n'
+                        f'# - see {url}'
+                    )
+                else:
+                    print(
+                        Fore.YELLOW +
+                        f'[{message["purpose"]}{message["index"]:03d}]'
+                        f'{file_}: {line_no}: {message["short"]}'
+                    )
+        if modify:
+            outlines.append(line)
+    if modify:
+        file_.write_text('\n'.join(outlines))
+    return count
+
+
+def get_cylc_files(base: Path) -> Generator[Path, None, None]:
+    """Given a directory yield paths to check.
+    """
+    for rglob in FILEGLOBS:
+        for path in base.rglob(rglob):
+            # Exclude log directory:
+            if not path.relative_to(base).parts[0].startswith('log'):
```

Comment:
I think we would want
```suggestion
            if path.relative_to(base).parts[0] != 'log':
```

Reply:
But this is the log directory name it's matching